Add Rancher upgrade note and MCM related changes for v1.2 (#405)
harvester/docs: docs/rancher/node/node-driver.md

- Changes made to the node driver configuration is not persisted. Any modifications applied will be reset upon restarting the Rancher container.
- Starting with Harvester node driver v0.6.3, the automatic injection of the `qemu-guest-agent` has been removed from the backend. If the image you are using does not contain the `qemu-guest-agent` package, you can still install it via the `userdata` config. Otherwise, the cluster will not be provisioned successfully.

   ```yaml
   #cloud-config
   package_update: true
   packages:
   - qemu-guest-agent
   runcmd:
   - - systemctl
     - enable
     - '--now'
     - qemu-guest-agent.service
   ```

:::

## RKE1 Kubernetes cluster
Click to learn [how to create RKE1 Kubernetes Clusters](./rke1-cluster.md).

## RKE2 Kubernetes cluster
Click to learn [how to create RKE2 Kubernetes Clusters](./rke2-cluster.md).

## K3s Kubernetes cluster
Click to learn [how to create k3s Kubernetes Clusters](./k3s-cluster.md).

## Topology spread constraints

_Available as of v1.0.3_

> Within your guest Kubernetes cluster, you can use [topology spread constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) to manage how workloads are distributed across nodes, accounting for factors such as failure domains like regions and zones. This can help to achieve high availability as well as efficient resource utilization of the Harvester cluster resources.

Reviewer: ```suggestion
Within your guest Kubernetes cluster, you can use [topology spread constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) to manage how workloads are distributed across nodes, accounting for factors such as failure domains like regions and zones. This helps achieve high availability and efficient resource utilization of the Harvester cluster resources.
```

> For RKE2 versions prior to `v1.25.x`, the minimum required versions to support the topology label sync feature are as follows:

Reviewer: ```suggestion
For RKE2 versions before `v1.25.x`, the minimum required versions to support the topology label sync feature are as follows:
```

| Minimum Required RKE2 Version |
| :--|
| \>=  v1.24.3+rke2r1 |
| \>=  v1.23.9+rke2r1 |
| \>=  v1.22.12+rke2r1 |

> Furthermore, for custom installation, the Harvester cloud provider version should be `>= v0.1.4`

Reviewer: ```suggestion
Furthermore, for custom installation, the Harvester cloud provider version should be `>= v0.1.4`.
```

### Sync topology labels to the guest cluster node

During the cluster installation, the Harvester node driver will automatically help synchronize topology labels from VM nodes to guest cluster nodes. Currently, only `region` and `zone` topology labels are supported.

> 1. Configure topology labels on the Harvester nodes on the **Hosts > Edit Config > Labels** page, for example, add the topology labels as follows:

Reviewer: ```suggestion
1. Configure topology labels on the Harvester nodes on the **Hosts** > **Edit Config** > **Labels** page. For example, add the topology labels as follows:
```
